wcivf/apps/elections/helpers.py

```diff
@@ -12,10 +16,56 @@ class EEHelper:
 
     ee_cache = {}
 
+    @property
+    def base_elections_url(self):
+        """
+        Builds the URL for the elections endpoint of EveryElection API
+        """
+        return f"{settings.EE_BASE}/api/elections/"
+
+    @cached_property
+    def deleted_election_ids(self):
+        """
+        Returns a generator object that retreives ids for elections that have
+        been deleted in Every Election. Accepts optional date arg, otherwise
+        defaults to objects with a Poll Open date within the last fifty days
+        """
+        params = {
+            "deleted": 1,
+            "poll_open_date__gte": datetime.now().date() - timedelta(days=50),
+        }
+        querystring = urlencode(params)
+        url = f"{self.base_elections_url}?{querystring}"
+        pages = JsonPaginator(page1=url, stdout=sys.stdout)
+        for page in pages:
+            return [result["election_id"] for result in page["results"]]
+
+    @transaction.atomic
+    def delete_deleted_elections(self):
+        """
+        Deletes Election and PostElection objects that were soft-deleted in
+        EveryElection
+        """
+        from elections.models import (
+            Election,
+            PostElection,
+        )  # noqa avoid circular import
+
+        elections_count, _ = Election.objects.filter(
+            slug__in=self.deleted_election_ids,
+            current=False,
```

Comment:
Can you explain this line? I don't _think_ we want to exclude `current` elections here...

Reply:
@symroe it was to match how YNR handles deleting elections https://github.com/DemocracyClub/yournextrepresentative/blob/bd0db75e46ef27efb2d2f459087f26f5e03cba21/ynr/apps/elections/models.py#L221

But if that's not important here I won't

Reply:
In fact I think we've already covered this in previous comments - so I'm going to update to remove the current check

Reply:
Oh that is interesting about EE->YNR imports - I'm not sure it's right in call cases! Either way, we care less here as none of the data is canonical in WCIVF.